components/config-mgmt-service/integration_test/node_run_test.go

```diff
@@ -61,13 +70,128 @@ func TestNodeRunWithRuns(t *testing.T) {
 		})
 }
 
+func TestNodeRunWithRunsNonmatchingNode(t *testing.T) {
+	var (
+		nodeID = newUUID()
+		run    = newIngestRun(nodeID, "success",
+			time.Now().Format(time.RFC3339),
+			time.Now().Add(time.Minute).Format(time.RFC3339))
+		req  = request.NodeRun{NodeId: run.EntityUuid, RunId: run.RunID}
+		ctx  = context.Background()
+		node = iBackend.Node{
+			Exists: true,
+			NodeInfo: iBackend.NodeInfo{
+				EntityUuid: "not-matching-run",
+			},
+		}
+	)
+
+	suite.IngestNodes([]iBackend.Node{node})
+	suite.IngestRuns([]iBackend.Run{run})
+	defer suite.DeleteAllDocuments()
+
+	_, err := cfgmgmt.GetNodeRun(ctx, &req)
+	assert.Error(t, err)
+}
+
+func TestNodeRunProjectFilter(t *testing.T) {
+	var (
+		nodeID = newUUID()
+		run    = newIngestRun(nodeID, "success",
+			time.Now().Format(time.RFC3339),
+			time.Now().Add(time.Minute).Format(time.RFC3339))
+		req                = request.NodeRun{NodeId: run.EntityUuid, RunId: run.RunID}
+		successfulResponse = ingestRunToMessage(run)
+	)
+
+	cases := []struct {
+		description   string
+		ctx           context.Context
+		nodeProjects  []string
+		expectedError bool
+	}{
+		{
+			description:   "Node project matching request projects",
+			ctx:           contextWithProjects([]string{"project9"}),
+			nodeProjects:  []string{"project9"},
+			expectedError: false,
+		},
+		{
+			description:   "Node project not matching request projects",
+			ctx:           contextWithProjects([]string{"project3"}),
+			nodeProjects:  []string{"project9"},
+			expectedError: true,
+		},
+		{
+			description:   "Node one project; request all projects allowed",
+			ctx:           contextWithProjects([]string{authzConstants.AllProjectsExternalID}),
+			nodeProjects:  []string{"project9"},
+			expectedError: false,
+		},
+		{
+			description:   "Node has no projects; request all projects allowed",
+			ctx:           contextWithProjects([]string{authzConstants.AllProjectsExternalID}),
+			nodeProjects:  []string{},
+			expectedError: false,
+		},
+		{
+			description:   "Node has no projects; request unassigned projects allowed",
+			ctx:           contextWithProjects([]string{authzConstants.UnassignedProjectID}),
+			nodeProjects:  []string{},
+			expectedError: false,
+		},
+		{
+			description:   "Node has a projects; request unassigned projects allowed",
```

Comment:
1. s/projects/project/ (here and on L150).
2. If I take your meaning correctly, it should also say s/allowed/not allowed/